guides/keras_nlp/upload.py

```diff
@@ -0,0 +1,238 @@
+"""
+"""
+### Upload to Kaggle Models
+"""
+
+"""
+To upload a model to Kaggle Models, first, we need to authenticate with Kaggle.
+This can by one of the followings:
+1. Set environment variables `KAGGLE_USERNAME` and `KAGGLE_KEY`.
+2. Provide a local `~/.kaggle/kaggle.json`.
+3. Call `kagglehub.login()`.
+
+Let's make sure we are logged in before coninuing.
+"""
+
+import kagglehub
+
+if "KAGGLE_USERNAME" not in os.environ or "KAGGLE_KEY" not in os.environ:
+    kagglehub.login()
+
+
+"""
+Run this command to upload the model that is save in `preset_dir` to Kaggle Models:
+"""
+
+kaggle_username = ""  # TODO: assing username.
+kaggle_uri = f"kaggle://{kaggle_username}/gpt2/keras/finetuned_gpt2"
+keras_nlp.upload_preset(kaggle_uri, preset_dir)
+
+"""
+### Upload to Hugging Face Hub
+"""
+
+"""
+To upload a model to Hugging Face Hub, first, we need to authenticate with Hugging Face.
+This can by one of the followings:
+1. Set environment variables `HF_USERNAME` and `HF_TOKEN`.
+2. Call `huggingface_hub.notebook_login()`.
+
+Let's make sure we are logged in before coninuing.
+"""
+
+import huggingface_hub
+
+if "HF_USERNAME" not in os.environ or "HF_TOKEN" not in os.environ:
+    huggingface_hub.notebook_login()
+
+"""
+Run this command to upload the model that is save in `preset_dir` to Hugging Face Hub:
+"""
+
+hf_username = huggingface_hub.whoami()
+hf_uri = f"hf://{hf_username}/finetuned_gpt2"
+keras_nlp.upload_preset(hf_uri, preset_dir)
+
+
+"""
+## Load a User Uploaded Model
+
+A user uploaded model can be loaded using `from_preset`.
+"""
+
+# Load a user uploaded CausalLM from Kaggle Models.
+causal_lm = keras_nlp.models.CausalLM.from_preset(
+    f"kaggle://{kaggle_username}/gpt2/keras/finetuned_gpt2"
+)
+
+# Load a user uploaded CausalLM from Hugging Face Hub.
+causal_lm = keras_nlp.models.CausalLM.from_preset(f"hf://{hf_username}/finetuned_gpt2")
+
+"""
+# Classifier Upload
+
+Uploading a classifier model is similar to Causal LM upload. You can follow
+[this guide](https://keras.io/guides/keras_nlp/getting_started/#fine-tuning-a-pretrained-bert-backbone)
+to fine-tune a classfier.
+
+
+To upload the fine-tuned model, first, the model is saved to a local directory using `save_to_preset`
+API and then it can be uploaded via `keras_nlp.upload_preset`.
+"""
+
+# Load the base model.
+classifier = keras_nlp.models.Classifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=2
+)
+
+# Fine-tune the classifier.
+classifier.fit(imdb_train)
+
+# Save the model to a local preset directory.
+preset_dir = "./finetuned_bert"
+classifier.save_to_preset(preset_dir)
+
+# Upload to Kaggle Models.
+keras_nlp.upload_preset(
+    f"kaggle://{kaggle_username}/bert/keras/finetuned_bert", preset_dir
+)
+
+# Load a user uploaded Classifier from Kaggle Models.
+classifier = keras_nlp.models.Classifier.from_preset(
+    f"kaggle://{kaggle_username}/bert/keras/finetuned_bert"
+)
+
+# Upload to Hugging Face.
```

Comment:
I don't think we need to show both hubs a second time. Probably fine to just stick with kaggle here.